diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7986906..64f2d0a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -169,7 +169,7 @@ elseif (TARGET SDL2)
 	endif()
 else()
 	message(STATUS "no TARGET SDL2::SDL2, or SDL2, using variables")
-	target_include_directories(FAudio PUBLIC "$<BUILD_INTERFACE:${SDL2_INCLUDE_DIRS}>")
+	target_include_directories(FAudio PUBLIC $<BUILD_INTERFACE:${SDL2_INCLUDE_DIRS}>)
 	target_link_libraries(FAudio PUBLIC ${SDL2_LIBRARIES})
 	if(INSTALL_MINGW_DEPENDENCIES)
 		install_shared_libs(bin ${SDL2_LIBRARIES})
