diff --git a/Cargo.toml b/Cargo.toml
index 115635a8..b5b10367 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -125,8 +125,8 @@ xpath = ["nu_plugin_xpath"]
 [profile.release]
 #strip = "symbols" #Couldn't get working +nightly
 codegen-units = 1 #Reduce parallel codegen units
-lto = true #Link Time Optimization
-opt-level = 'z' #Optimize for size
+lto = "thin" #Link Time Optimization
+opt-level = 3 #Optimize for size
 
 # Core plugins that ship with `cargo install nu` by default
 # Currently, Cargo limits us to installing only one binary
