diff --git a/dlls/winex11.drv/opengl.c b/dlls/winex11.drv/opengl.c
index 7fb97f5..4b1c67b 100644
--- a/dlls/winex11.drv/opengl.c
+++ b/dlls/winex11.drv/opengl.c
@@ -861,7 +861,7 @@ static int ConvertAttribWGLtoGLX(const int* iWGLAttr, int* oGLXAttr, struct wgl_
           switch (pop)
           {
           case WGL_SWAP_EXCHANGE_ARB:
-              pop = GLX_SWAP_EXCHANGE_OML;
+              pop = GLX_DONT_CARE;
               break;
           case WGL_SWAP_COPY_ARB:
               pop = GLX_SWAP_COPY_OML;
